src/Assignments/SimpleProblemPage.tsx

```diff
@@ -91,12 +91,21 @@ export const SimpleProblemPage: React.FC<SimpleProblemPageProps> = () => {
                 !_.isNil(topic.topicAssessmentInfo.studentTopicAssessmentInfo) &&
                 topic.topicAssessmentInfo.studentTopicAssessmentInfo.length > 0 // student has generated at least one version already
             ) {
-                topic.topicAssessmentInfo.studentTopicAssessmentInfo = _.sortBy(topic.topicAssessmentInfo.studentTopicAssessmentInfo, ['startTime'], ['desc']);
-                setAttemptsRemaining(topic.topicAssessmentInfo.studentTopicAssessmentInfo[0].maxAttempts! - topic.topicAssessmentInfo.studentTopicAssessmentInfo[0].numAttempts!);
+                const currentVersion = _.maxBy(topic.topicAssessmentInfo.studentTopicAssessmentInfo, 'startTime');
+                if (!_.isNil(currentVersion) && !_.isNil(currentVersion?.numAttempts) && !_.isNil(currentVersion.maxAttempts)) {
+                    // if the assessment has expired - do NOT allow submissions...
```

Comment:
We are blocking this on the backend as well right?

Reply:
yes, but I'd rather not put the students in an error state

Reply:
💯